### PR TITLE
5229 request access api

### PR DESCRIPTION
**Pull request by sekmiller**
## New Contributors

Welcome! New contributors should at least glance at [CONTRIBUTING.md](/CONTRIBUTING.md), especially the section on pull requests where we encourage you to reach out to other developers before you start coding. Also, please note that we measure code coverage and prefer you write unit tests. Pull requests can still be reviewed without tests or completion of the checklist outlined below. Thanks!

## Related Issues

- connects to #5229 Request Access API

## Pull Request Checklist

- [x] Unit [tests][] none
- [x] Integration [tests][x]: completed
- [x] Deployment requirements, [x] None
- [x] [Documentation][docs] [x]completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: src/test/java/edu/harvard/iq/dataverse/api/AccessIT.java [hunk 1]
```diff
@@ -402,5 +397,68 @@ public void testDownloadMultipleFiles_LoggedAndNot_Unpublished() throws IOExcept
...
+        //grant file access
+        Response grantFileAccessResponse = UtilIT.grantFileAccess(tabFile3IdRestrictedNew.toString(), "@" + apiIdentifierRando, apiToken);
+        assertEquals(200, grantFileAccessResponse.getStatusCode());
+
+        //revokeFileAccess        
+        Response revokeFileAccessResponse = UtilIT.revokeFileAccess(tabFile3IdRestrictedNew.toString(), "@" + apiIdentifierRando, apiToken);
+        assertEquals(200, revokeFileAccessResponse.getStatusCode());
+        
+        listAccessRequestResponse = UtilIT.getAccessRequestList(tabFile3IdRestrictedNew.toString(), apiToken);
+        assertEquals(400, listAccessRequestResponse.getStatusCode());
```
[the remark this answers is not in the document]
Comment (sekmiller) [reply src/test/java/edu/harvard/iq/dataverse/api/AccessIT.java]: Done

File: src/main/java/edu/harvard/iq/dataverse/api/Access.java [hunk 2]
```diff
@@ -880,6 +914,391 @@ private String getWebappImageResource(String imageName) {
...
+     *
+     * @author sekmiller
+     *
+     * @param fileToRequestAccessId
+     * @param identifier
+     * @param apiToken
+     * @param headers
+     * @return
+     */
+    @GET
```
[the remark this answers is not in the document]
Comment (sekmiller) [reply src/main/java/edu/harvard/iq/dataverse/api/Access.java]: Done

File: src/main/java/edu/harvard/iq/dataverse/api/Access.java [hunk 3]
```diff
@@ -880,6 +914,391 @@ private String getWebappImageResource(String imageName) {
...
```
[the remark this answers is not in the document]
Comment (sekmiller) [reply src/main/java/edu/harvard/iq/dataverse/api/Access.java]: Done

File: src/main/java/edu/harvard/iq/dataverse/api/Access.java [hunk 4]
```diff
@@ -880,6 +914,391 @@ private String getWebappImageResource(String imageName) {
...
+     * Request Access to Restricted File
+     *
+     * @author sekmiller
+     *
+     * @param fileToRequestAccessId
+     * @param apiToken
+     * @param headers
+     * @return
+     */
+    @GET
```
[the remark this answers is not in the document]
Comment (sekmiller) [reply src/main/java/edu/harvard/iq/dataverse/api/Access.java]: Done

File: src/main/java/edu/harvard/iq/dataverse/api/Access.java [hunk 5]
```diff
@@ -878,6 +912,391 @@ private String getWebappImageResource(String imageName) {
...
+
+        try {
+            dataFile = findDataFileOrDie(fileToRequestAccessId);
+        } catch (WrappedResponse ex) {
+            List<String> args = Arrays.asList(fileToRequestAccessId);
+            return error(BAD_REQUEST, BundleUtil.getStringFromBundle("access.api.requestAccess.fileNotFound", args));
+        }
+
+        if (apiToken == null || apiToken.equals("")) {
+            apiToken = headers.getHeaderString(API_KEY_HEADER);
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Access.java]: Switch to findUserOrDie and remove key.

File: src/main/java/edu/harvard/iq/dataverse/api/Access.java [hunk 6]
```diff
@@ -878,6 +912,391 @@ private String getWebappImageResource(String imageName) {
...
+        
+        try {
+            dataFile = findDataFileOrDie(fileToRequestAccessId);
+        } catch (WrappedResponse ex) {
+            List<String> args = Arrays.asList(fileToRequestAccessId);
+            return error(BAD_REQUEST, BundleUtil.getStringFromBundle("access.api.requestAccess.fileNotFound", args));
+        }
+        
+        if (apiToken == null || apiToken.equals("")) {                   
+            apiToken = headers.getHeaderString(API_KEY_HEADER);
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Access.java]: Remove key. Switch to findUserOrDie.

**Comment by kcondon** [comment]
removed connection to 2855 at @mheppler's request due to not addressing that issue.

**Pull request merged**